### PR TITLE
create spark_connect options for RAPIDS accelerator plugin

### PR DESCRIPTION
**Pull request by yl790**
@falaki  I noticed there also appears to be some Databricks-specific fork of rapids-4-spark_2.12, so presumably that means we shall have the following:

```
    packages <- c(
      packages,
      (
        if (method %in% c("databricks", "databricks-connect"))
          "com.nvidia:rapids-4-spark_2.12:0.1.0-databricks"
        else
          "com.nvidia:rapids-4-spark_2.12:0.1.0"
      ),
      "ai.rapids:cudf:0.14"
    )
```

Signed-off-by: yl790 <yitao@rstudio.com>

### CODE REVIEW

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-1/b3ef1d33-8e6d-48aa-af2e-6f5bcaea9b1a.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-2/60d8c9ad-02f1-49e2-940e-8cdff4155e63.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-3/3e7ba206-fd7d-4475-894e-eb9aa4887d2d.txt).

**Comment by yl790** [comment]
Was finally able to see this in action as well (but only after getting a p3.2xlarge gpu instance on EC2 as all other smaller instances were unsupported):

```
library(sparklyr)

sc <- spark_connect(master = "local", version = "3.0.0", enable_spark_rapids_plugin = TRUE)
DBI::dbGetQuery(sc, "EXPLAIN SELECT 4;")

##  plan
## 1 == Physical Plan ==\n*(2) GpuColumnarToRow false\n+- GpuProject [4 AS 4#5]\n   +- GpuRowToColumnar TargetSize(2147483647)\n      +- *(1) Scan OneRowRelation[]\n\n
```

We should most definitely have at least 1 example showing perf improvement from the RAPIDS library with some actual sparklyr use case as well. No need to do that now though.

File: R/shell_connection.R [hunk 1]
```diff
@@ -20,7 +20,9 @@ shell_connection <- function(master,
                              remote,
                              extensions,
                              batch,
-                             scala_version = NULL) {
+                             scala_version = NULL,
+                             enable_spark_rapids_plugin = FALSE,
```
Comment (javierluraschi) [inline R/shell_connection.R:24]: I think we rather want `rapids` to be specified in `packages`, as in:

```r
sc <- spark_connect(master = "local", packges = "rapids")
```

And also to specify the configuration in `spark_config` with the `rapids.` prefix, as in:

```r
config <- spark_config()
config$rapids.some.key = "value"
config$rapids.some.other = "other"
```

And then extract the config options, remove the prefix and use as `spark_rapids_plugin_conf` internally.

**Comment by falaki** [comment]
Should we add any tests for this? As for the Databricks specific JARs, I believe they are maintained and published by Nvidia. I will ask my Nvidia contacts to take a look at this PR.

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-4/4d3d3aaf-ad2c-42e8-9220-a163ade26699.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-5/48ec17fd-d579-48a6-aaa0-927686a3fff0.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-6/7e93b902-aa06-4466-b9ac-d4bf1fec4975.txt).

**Comment by yl790** [comment]
Note to myself: after this change

```
list(
  sparklyr.shell.conf = "key1=value1",
  sparklyr.shell.conf = "key2=value2"
)
```

will translate to `--conf key1=value1 --conf key2=value2` in spark-submit args (expected behavior)

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-8/b7662bc3-635f-46b5-85a4-c104a6f3b1e4.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-7/b462fa6c-094b-4167-973f-e81953660043.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-9/12f0e3fa-900f-4d6b-8ba9-5729139e8282.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-10/9fa81100-c76a-4802-b66e-2551082af982.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-11/b8bd6faa-fa0e-4bbd-8c35-b108665d72fd.txt).

**Comment by falaki** [comment]
Databricks Connect tests failed. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-14/874a8d5a-d808-405b-a76e-babb5158e05a.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-12/435d6f57-7700-43b1-93f6-28165dcb76b9.txt).

**Comment by falaki** [comment]
Databricks Connect tests succeeded. View logs [here](https://sparklyr-jenkins.s3.amazonaws.com/logs/jenkins-sparklyr-PR-2632-13/40f0d061-d14a-40e5-8484-cd9b12bc6559.txt).

**Pull request merged**